### ISSUE TITLE
[Accessibility]: Code Insights - Insight creation card doesn't announce insight type 

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#41793]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33493

### Problem description

When you focus "Create" button within one of the insight creation card voice over doesn't announce type of insight that you're about to create. 

### Expected behavior

When you focus create insight button it should announce type of insight like
"Create Track and detect insight".

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
Need to make sure that "Create" is still contained in the longer `aria-label` to not violate 2.5.3: Label in Name.
